common/theme_generator.py

```diff
@@ -70,10 +70,13 @@ def get_new_theme_path(self, name):
 
         full_path = os.path.join(sublime.packages_path(), path_in_packages)
 
-        with open(full_path, "wb") as out_f:
-            out_f.write(STYLES_HEADER.encode("utf-8"))
-            out_f.write(ElementTree.tostring(self.plist, encoding="utf-8"))
-
+        try:
+            with open(full_path, "wb") as out_f:
```

Comment:
My apologies, but on second look, it would be better to resolve the current issue as-is and later, introduce the full change a separate pull request.

Regardless, in Python, the `contextlib.contextmanager` decorator makes this sort of thing trivial (please note that I have not tested this code):

```python
from contextlib import contextmanager

import sublime


@contextmanager
def safe_open(filename, mode, *args, **kwargs):
    try:
        with open(filename, mode, *args, **kwargs) as file:
            yield file
    except PermissionError as e:
        sublime.ok_cancel_dialog("GitSavvy could not access file: \n{}".format(e))
        raise e
    except OSError as e:
        sublime.ok_cancel_dialog("GitSavvy encountered an OS error: \n{}".format(e))
        raise e
```

This creates a new context manager that can be anywhere open can:

```python
def get_new_theme_path(self, name):

    # ... other code

    with safe_open(full_path, "wb") as out_f:
        out_f.write(STYLES_HEADER.encode("utf-8"))
        out_f.write(ElementTree.tostring(self.plist, encoding="utf-8"))
```